js/multiFileUpload.js

```diff
@@ -1,351 +1,390 @@
-(function ($)
+(function ($) {
+
+  "use strict";
+
+  Qubit.multiFileUpload = Qubit.multiFileUpload || {};
+
+  var MultiFileUpload = function (element)
   {
-    Qubit.multiFileUpload = Qubit.multiFileUpload || {};
-
-    Drupal.behaviors.multiFileUpload = {
-      attach: function (context)
-        {
-          // Enable uploader
-          var uploader = new YAHOO.widget.Uploader('uploaderOverlay');
-
-          // Put swf object over "Select files" button
-          var uiLayer = YAHOO.util.Dom.getRegion('selectLink');
-          var overlay = $('#uploaderOverlay');
-          overlay.width(uiLayer.right - uiLayer.left);
-          overlay.height(uiLayer.bottom - uiLayer.top);
-
-          function parseHtmlId(id)
-          {
-            var yuiId = id.match(/([0-9]+)$/).shift();
-
-            if (null != yuiId && !isNaN(parseInt(yuiId)))
-            {
-              return yuiId;
+    this.uppy = new Uppy.Core({ 
+      debug: false,
+      id: 'Uppy-atom',
+      autoProceed: false,
+      restrictions: {
+        minNumberOfFiles: 1
+      },
+      onBeforeFileAdded: (currentFile, files) => this.onBeforeFileAddedChecks(currentFile, files),
+    });
+
+    this.nextImageNum = 1;
+    this.uploadItems = [];
+    this.result = "";
+
+    this.$element = $(element);
+    this.$submitButton = this.$element.find('input[type="submit"]');
+    this.$cancelButton = this.$element.find('a[title="Cancel"]');
+    this.$retryButton = $('<a class="c-btn" title="retry"/>')
+      .attr('type','hidden')
+      .text(Qubit.multiFileUpload.i18nRetry)
+      .appendTo('.actions');
+
+    this.init();
+    this.listen();
+  };
+
+  MultiFileUpload.prototype = {
+    constructor: MultiFileUpload,
```

Comment:
We won't need this if we follow the comment above, as it's the default behavior when a function is extended with prototype.

Reply:
Updated as above.